src/plugins/explore/public/application/utils/state_management/actions/query_actions.ts

```diff
@@ -0,0 +1,539 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { i18n } from '@osd/i18n';
+import moment from 'moment';
+import { IBucketDateHistogramAggConfig, Query, DataView } from 'src/plugins/data/common';
+import { QueryExecutionStatus } from '../types';
+import { setResults, ISearchResult } from '../slices';
+import { setIndividualQueryStatus } from '../slices/query_editor/query_editor_slice';
+import { ExploreServices } from '../../../../types';
+import {
+  DataPublicPluginStart,
+  indexPatterns as indexPatternUtils,
+  search,
+} from '../../../../../../data/public';
+import {
+  buildPointSeriesData,
+  createHistogramConfigs,
+  getDimensions,
+} from '../../../../components/chart/utils';
+import { SAMPLE_SIZE_SETTING } from '../../../../../common';
+import { RootState } from '../store';
+import { getResponseInspectorStats } from '../../../../application/legacy/discover/opensearch_dashboards_services';
+import {
+  ChartData,
+  DefaultDataProcessor,
+  HistogramDataProcessor,
+  ProcessedSearchResults,
+} from '../../interfaces';
+import { defaultPreparePplQuery, getQueryWithSource } from '../../languages';
+
+// Module-level storage for abort controllers keyed by cacheKey
+const activeQueryAbortControllers = new Map<string, AbortController>();
+
+// Helper function to abort all active queries
+export const abortAllActiveQueries = () => {
+  activeQueryAbortControllers.forEach((controller, cacheKey) => {
+    controller.abort();
+  });
+  activeQueryAbortControllers.clear();
+};
+
+/**
+ * Default query preparation for tabs
+ */
+export const defaultPrepareQueryString = (query: Query): string => {
```

Comment:
this function seems really complex and not sure if it's helpful. right now it seems like the only purpose for this function is to strip stats from the query but then that gets obfuscated based on the name